bleak/backends/bluezdbus/client.py

```diff
@@ -53,27 +53,45 @@ def __init__(self, address_or_ble_device: Union[BLEDevice, str], **kwargs):
         # kwarg "device" is for backwards compatibility
         self._adapter = kwargs.get("adapter", kwargs.get("device", "hci0"))
 
-        # Backend specific, TXDBus objects and data
+        # Backend specific, D-Bus objects and data
         if isinstance(address_or_ble_device, BLEDevice):
             self._device_path = address_or_ble_device.details["path"]
             self._device_info = address_or_ble_device.details.get("props")
         else:
             self._device_path = None
             self._device_info = None
-        self._bus = None
-        self._reactor = None
-        self._rules = {}
-        self._subscriptions = list()
-
-        # This maps DBus paths of GATT Characteristics to their BLE handles.
-        self._char_path_to_handle = {}
 
-        # We need to know BlueZ version since battery level characteristic
-        # are stored in a separate DBus interface in the BlueZ >= 5.48.
+        # D-Bus message bus
+        self._bus: Optional[MessageBus] = None
+        # match rules we are subscribed to that need to be removed on disconnect
+        self._rules: List[MatchRules] = []
+        # D-Bus properties for the device
+        self._properties: Dict[str, Any] = {}
+        # list of characteristic handles that have notifications enabled
+        self._subscriptions: List[int] = []
+        # provides synchronization between get_services() and PropertiesChanged signal
+        self._services_resolved_event: Optional[asyncio.Event] = None
+        # indicates disconnect request in progress when not None
+        self._disconnecting_event: Optional[asyncio.Event] = None
+        # used to ensure device gets disconnected if event loop crashes
+        self._disconnect_monitor_event: Optional[asyncio.Event] = None
+
+        # get BlueZ version
         p = subprocess.Popen(["bluetoothctl", "--version"], stdout=subprocess.PIPE)
         out, _ = p.communicate()
         s = re.search(b"(\\d+).(\\d+)", out.strip(b"'"))
-        self._bluez_version = tuple(map(int, s.groups()))
+        bluez_version = tuple(map(int, s.groups()))
+
+        # BlueZ version features
+        self._can_write_without_response = (
+            bluez_version[0] == 5 and bluez_version[1] >= 46
+        )
+        self._write_without_response_workaround_needed = (
+            bluez_version[0] == 5 and bluez_version[1] < 51
+        )
+        self._hides_battery_characteristic = self._hides_device_name_characteristic = (
```

Comment:
I opened #393 to address this separately. I am hoping that we can change it to some logic like "if the service is not found, then check the bus for a Battery1 object" rather than looking at the BlueZ version.